models/template-version.js

```diff
@@ -0,0 +1,147 @@
+import {
+  sparqlEscapeUri,
+  uuid as uuidv4,
+  sparqlEscapeDateTime,
+  sparqlEscapeString,
+} from "mu";
+import { querySudo as query, updateSudo as update } from "@lblod/mu-auth-sudo";
+import fs from "fs";
+
+export default class TemplateVersion {
+  /** @type {string} */
+  id;
+  /** @type {string} */
+  uri;
+  /** @type {title} */
+  title;
+  /** @type {string} */
+  derivedFrom;
+
+  /**
+   *
+   * @param {{uri: string, id: string, title: string, derivedFrom: string}}
+   */
+  constructor({ uri, id, title, derivedFrom }) {
+    this.uri = uri;
+    this.id = id;
+    this.title = title;
+    this.derivedFrom = derivedFrom;
+  }
+
+  /**
+   *
+   * @param {{ derivedFrom: string, title: string, content: string }}
+   */
+  static async create({ derivedFrom, title, content }) {
+    const now = new Date();
+    const fileUuid = uuidv4();
+    const fileName = `${fileUuid}.html`;
+    const filePath = `/share/${fileName}`;
+    fs.writeFileSync(filePath, content);
+    const fileSize = fs.statSync(filePath).size;
+    const physicalFileUuid = uuidv4();
```

Comment:
Is there a reason to generate a different UUID to represent the file in the triplestore as is used on the filesystem?

Reply:
I took over this code from the original code, it would probably make sense that the filename uses the same id as the physical-file resource.